File: ios-native/NeobankNative/Sources/Features/Auth/AuthModels.swift
import Foundation

struct AuthTokens: Decodable, Sendable {
    let userId: String
    let accessToken: String
    let refreshToken: String
}

struct Profile: Decodable, Sendable {
    let userId: String
    let email: String
    let phone: String
    let status: String
    let kycStatus: String
    let createdAt: Date
    let fullName: String?
    let dateOfBirth: String?
    let countryCode: String?
}

File: android-native/app/src/main/kotlin/dev/horobets/neobanknative/features/auth/AuthModels.kt
package dev.horobets.neobanknative.features.auth

import kotlinx.serialization.Serializable

@Serializable
data class AuthTokens(
    val userId: String,
    val accessToken: String,
    val refreshToken: String,
)

@Serializable
data class Profile(
    val userId: String,
    val email: String,
    val phone: String,
    val status: String,
    val kycStatus: String,
    val createdAt: String,
    val fullName: String? = null,
    val dateOfBirth: String? = null,
    val countryCode: String? = null,
)
